prometheus_juju_exporter/collector.py

```diff
@@ -137,14 +129,19 @@ async def _get_machine_stats(
         :param str model_name: the name of the model the machines are in
         :param str gauge_name: the name of the gauge
         """
+        virt_mac_prefixes = self.config["machine"]["virt_macs"].get(str).split(",")
+
         for machine in machines.values():
             # check machine type with the prefix of its mac address
             interfaces = machine["network-interfaces"].values()
             machine_type = MachineType.METAL
 
             for i in interfaces:
                 mac_addresses = i["mac-address"]
-                if mac_addresses.startswith(tuple(self.prefixes)):
+                if (
+                    mac_addresses.startswith(tuple(virt_mac_prefixes))
```

Comment:
why plural? `mac_addresses` will only contain one mac address, right?

Reply:
This variable used to store a list of mac addresses and I didn't rename it after having changed it to store a single MAC at a time. I've updated it.